server/service/src/auth.rs

```diff
@@ -357,7 +366,7 @@ fn validate_resource_permissions(
                     resource_request,
                     child,
                 ) {
-                    ()
+                    return Ok(());
```

Comment:
This is the only code change, return the result as soon as any one permission is matched.